data-prepper-core/src/main/java/com/amazon/dataprepper/peerforwarder/PeerForwarderConfiguration.java

```diff
@@ -0,0 +1,181 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.amazon.dataprepper.peerforwarder;
+
+import com.amazon.dataprepper.peerforwarder.discovery.DiscoveryMode;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import jakarta.validation.constraints.AssertTrue;
+
+import java.nio.file.Files;
+import java.nio.file.Paths;
+
+/**
+ * Class to hold configuration for Core Peer Forwarder in {@link com.amazon.dataprepper.parser.model.DataPrepperConfiguration},
+ * including port, ssl, buffer, peer forwarder client and server configuration.
+ * @since 2.0
+ */
+
+public class PeerForwarderConfiguration {
+    private Integer serverPort = 21890;
+    private Integer requestTimeout = 10_000;
+    private Integer threadCount = 200;
+    private Integer maxConnectionCount = 500;
+    private Integer maxPendingRequests = 1024;
+    private Boolean ssl = true;
+    private String sslCertificateFile;
+    private String sslKeyFile;
+    private DiscoveryMode discoveryMode;
+    private Integer batchSize = 48;
+    private Integer bufferSize = 512;
+
+    public PeerForwarderConfiguration() {}
+
+    @JsonCreator
+    public PeerForwarderConfiguration (
+            @JsonProperty("port") final Integer serverPort,
+            @JsonProperty("request_timeout") final Integer requestTimeout,
+            @JsonProperty("thread_count") final Integer threadCount,
+            @JsonProperty("max_connection_count") final Integer maxConnectionCount,
+            @JsonProperty("max_pending_requests") final Integer maxPendingRequests,
+            @JsonProperty("ssl") final Boolean ssl,
+            @JsonProperty("ssl_certificate_file") final String sslCertificateFile,
+            @JsonProperty("ssl_key_file") final String sslKeyFile,
+            @JsonProperty("discovery_mode") final String discoveryMode,
+            @JsonProperty("batch_size") final Integer batchSize,
+            @JsonProperty("buffer_size") final Integer bufferSize
+    ) {
+        setServerPort(serverPort);
+        setRequestTimeout(requestTimeout);
+        setThreadCount(threadCount);
+        setMaxConnectionCount(maxConnectionCount);
+        setMaxPendingRequests(maxPendingRequests);
+        setSsl(ssl);
+        setDiscoveryMode(discoveryMode);
+        this.sslCertificateFile = sslCertificateFile != null ? sslCertificateFile : "";
+        this.sslKeyFile = sslKeyFile != null ? sslKeyFile : "";
+        setBatchSize(batchSize);
+        setBufferSize(bufferSize);
+    }
+
+    @AssertTrue(message = "ssl_certificate_file must be a valid file path when ssl is enabled")
```

Comment:
I'm not sure that Java validation is wired up when using Spring to configure these. So we might need to configure Spring to support it, and possibly add an annotation in the AppConfig. I'm not completely sure right now how to do it, or if it is necessary. This might need to go beyond the scope of this PR though. We need to track it.

Reply:
For now, I can create a setter for these 2 fields and throw `IllegalArgumentException`.